### PR TITLE
Established server connection & added application dependencies.

### PR DESCRIPTION
**Pull request by EmadBinAbid**
Issue#1 solved.
Issue#2 solved. (mongoose  version 5.1.7 used.)
Issue#3 solved. (jwt authentication used.)
Issue#4 solved. 
Issue#5 solved.

### CODE REVIEW

File: authentication.mongoose.js [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+    {
+        UserModel.findOne(req.body.user, (err, dbObject) => 
+        {
+            if(err)
+            {
+                res.status(401).send('Unauthorized');
+            }
+            else
+            {
+                if(dbObject === null)
```
Comment (AhsanAyaz) [inline authentication.mongoose.js]: rename `dbObject` to `user` maybe. better to keep variable names such that they reflect what they are

File: models/event-model/event.router.js [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+                    {
+                        res.status(400).send("Bad request");
+                    }
+                    else
+                    {
+                        res.json({ "event": dbObject });
+                    }
+                });
+                
+                // EventModel.create(req.body, (err, dbObject) => {
```
Comment (AhsanAyaz) [inline models/event-model/event.router.js]: remove the commented code

File: models/event-model/event.router.js [hunk 3]
```diff
@@ -0,0 +1,182 @@
...
+            if(err)
+            {
+                res.status(401).send("Unauthorized");
+            }
+            else
+            {
+                const query = { _id: req.query._id };
+                const options = { new: true };
+
+                EventModel.findOneAndUpdate(query, req.body.event, options, (err, dbObject) => {
```
Comment (AhsanAyaz) [inline models/event-model/event.router.js]: change `dbObject` to `event`

**Comment by EmadBinAbid** [comment]
Requested changes made successfully.

**Pull request merged**